conans/client/build/compiler_id.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+
+import os
+import tempfile
+from six import StringIO
+
+from conans.client.runner import ConanRunner
+
+
+GCC = "gcc"
+LLVM_GCC = "llvm-gcc"  # GCC frontend with LLVM backend
+CLANG = "clang"
+APPLE_CLANG = "apple-clang"
+SUNCC = "suncc"
+MSVC = "Visual Studio"
+INTEL = "intel"
+QCC = "qcc"
+
+
+class CompilerId:
+    """
+    compiler identification description, holds compiler name, full version and
+    other important properties
+    """
+    def __init__(self, name, major, minor, patch):
+        self._name = name
+        self._major = major
+        self._minor = minor
+        self._patch = patch
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def major(self):
+        return self._major
+
+    @property
+    def minor(self):
+        return self._minor
+
+    @property
+    def patch(self):
+        return self._patch
+
+    @property
+    def version(self):
+        return "%s.%s.%s" % (self._major, self._minor, self._patch)
+
+    @property
+    def major_minor(self):
+        return "%s.%s" % (self._major, self._minor)
+
+    def __str__(self):
+        return "%s %s" % (self._name, self.version)
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __eq__(self, other):
+        return self.name == other.name and \
+               self.major == other.major and \
+               self.minor == other.minor and \
+               self.patch == other.patch
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+
+UNKWNON_COMPILER = CompilerId(None, None, None, None)
+
+MSC_VER = {800: (1, 0),
```

Comment:
Please call this something like ``MSVC_TO_VS_VERSION`` and add some comment about the meaning of the numbers

Reply:
Or ``MSC_VER_TO_VS``

Reply:
renamed